pgmpy/readwrite/XMLBIF.py

```diff
@@ -233,20 +242,24 @@ def __str__(self):
         """
         if self.prettyprint:
             self.indent(self.xml)
-        return etree.tostring(self.xml, encoding=self.encoding)
+        f = BytesIO()
+        et = etree.ElementTree(self.xml)
+        et.write(f, encoding='utf-8', xml_declaration=True)
+        return f.getvalue().decode('utf-8')  # your XML file, encoded as UTF-8
+        # return etree.tostring(self.xml, encoding=self.encoding)
```

Comment:
@finnhacks42 This needs a explanation. What was the issue with previous behaviour ? Also why we are always returning XML file with UTF-8 encoding if we have encoding parameter.